unstructured/cleaners/translate.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Optional
+import warnings
+
+import langdetect
+from transformers import MarianMTModel, MarianTokenizer
+
+from unstructured.staging.huggingface import chunk_by_attention_window
+from unstructured.nlp.tokenize import sent_tokenize
+
+
+def _get_opus_mt_model_name(source_lang: str, target_lang: str):
+    """Constructs the name of the MarianMT machine translation model based on the
+    source and target language."""
+    return f"Helsinki-NLP/opus-mt-{source_lang}-{target_lang}"
+
+
+def _validate_language_code(language_code: str):
+    if not isinstance(language_code, str) or len(language_code) != 2:
+        raise ValueError(
+            f"Invalid language code: {language_code}. Language codes must be " "two letter strings."
+        )
+
+
+def translate_text(text, source_lang: Optional[str] = None, target_lang: str = "en"):
+    """Translates the foreign language text. Uses the stage
+
+    Parameters
+    ----------
+    text: str
+        The text to translate
+    target_lang: str
+        The two letter language code for the target langague. Defaults to "en".
+    source_lang: Optional[str]
+        The two letter language code for the language of the input text. If source_lang is
+        not provided, the function will try to detect it.
+    """
+    if text.strip() == "":
+        raise ValueError("An empty string was passed to translate_text.")
```

Comment:
What are the concerns around trying to translate an empty string? Is it just that we think this was a mistake, or does this cause problems if we allow it to go into the models?

Reply:
Empty strings were causing a problem with `langdetect` because there string didn't have any features it could use to detect the language. Realized we don't really need to run the translation though if it's an empty string. Now just returning the original text if it's empty.